### DISCUSSION TITLE
Debugging/troubleshooting?

### DISCUSSION THREAD

**Discussion by j4nd3r53n** [TPC-Council/HammerDB#273] [Q&A] [answered]
Is there any way to get hammerdbcli to produce detailed output - like a debug flag? I am trying to figure out why hammerdb seems to stall when I try to use it on our AWS Aurora instances - it seems to work fine with classic MySQL, so there must be something wrong elsewhere. Unfortunately, Aurora serverless is simply a black box - there is no way to extract information about what is going on on a deeper level than what AWS have decided to publish.

Any help would be appreciated.

**Comment by sm-shaw** [comment, marked as answer]
Issue converted to discussion.
Yes, the 'test' driver script is exactly for this functionality.  In the GUI there is a radio button option to choose between them and in the CLI you would do it as shown:
```
hammerdb>diset tpcc mysql_driver timed
Clearing Script, reload script to activate new setting
Script cleared
Changed tpcc:mysql_driver from test to timed for MySQL

hammerdb>diset tpcc mysql_driver test
Clearing Script, reload script to activate new setting
Script cleared
Changed tpcc:mysql_driver from timed to test for MySQL
```
The test driver script is not intended for a full workload test, but instead runs all Virtual Users, displaying the output. This is used for troubleshooting with one or a small number of virtual users to make sure the workload is running as expected. (Also in GUI environments displays are not multi-threaded so the output needs to be passed to one thread to be printed thereby introducing a bottleneck).

In both the test and timed workloads there is the RAISE ERROR option which by default is false. If an error is detected by default it will be caught and ignored. Setting this to true will cause the ERROR to be caught printed and the Virtual User to error - so this is the next level of troubleshooting.

```
hammerdb>diset tpcc mysql_raiseerror true
Changed tpcc:mysql_raiseerror from false to true for MySQL
```

Finally, as everything is scripted it is straightforward to modify the driver scripts to print as much information as you need.
